File: SoftUni Lessons/Python Development/Python Fundamentals September 2019/Problems And Files/27 EXERCISE TEXT PROCESSING - Дата 15-ти ноември, 1430 - 1730/03. Extract File.py
"""
Text Processing - Exercise
Check your code: https://judge.softuni.bg/Contests/Practice/Index/1740#2

SUPyF2 Text-Pr.-Ex. - 03. Extract File

Problem:
Write a program that reads the path to a file and subtracts the file name and its extension.

Examples:
Input:
C:\Internal\training-internal\Template.pptx
Output:
File name: Template
File extension: pptx

Input:
C:\Projects\Data-Structures\LinkedList.cs
Output:
File name: LinkedList
File extension: cs
"""
text = input()[::-1]
new_text = ""

for letter in text:
    if letter == "\\":
        break
    new_text += letter
file, extension = new_text[::-1].split(".")
print(f"File name: {file}")
print(f"File extension: {extension}")

File: SoftUni Lessons/JavaScript Development/JS Fundamental REDO 2022/23 - Text Processing - Exercise/03. Extract File.js
function extractFile (path) {
    // MasK
    let file = path.split('\\')[path.split('\\').length - 1].split('.');
    let fileName = file.slice(0, file.length -1).join('.');
    let fileExt = file[file.length - 1];

    console.log(`File name: ${fileName}`);
    console.log(`File extension: ${fileExt}`);
}

extractFile('C:\\Internal\\training-internal\\Template.pptx');